### PR TITLE
NXP-21124 fix packaging using graph resolution instead of ad-hoc

### PR DESCRIPTION
**Pull request by dmetzler**

### CODE REVIEW

**Comment by ataillefer** [comment]
:+1:

**Comment by efge** [comment]
👍

File: marketplace/src/main/assemble/assembly.xml [hunk 1]
```diff
@@ -1,41 +1,142 @@
...
-  <taskdef resource="org/nuxeo/build/antlib.xml" uri="urn:nuxeo-build" />
-  <taskdef resource="org/nuxeo/build/artifact/antlib.xml" uri="urn:nuxeo-artifact" />
+  <taskdef resource="org/nuxeo/build/antlib.xml" uri="urn:nuxeo-build"/>
+  <taskdef resource="org/nuxeo/build/artifact/antlib.xml" uri="urn:nuxeo-artifact"/>
+  <taskdef resource="net/sf/antcontrib/antlib.xml"/>
 
-  <property name="outdir" value="${maven.project.build.directory}" />
+  <target name="build" description="Build Nuxeo Package">
+    <property name="outdir" value="${maven.project.build.directory}"/>
+    <!-- Include your artifacts in the graph adding your groupId to groupPrefixes -->
```
Comment (jcarsique) [inline marketplace/src/main/assemble/assembly.xml]: please remove comments addressed to customers copying the sample

File: marketplace/src/main/assemble/assembly.xml [hunk 2]
```diff
@@ -1,41 +1,142 @@
...
+        <includes>
+          <artifact groupId="!org.nuxeo*" />
+          <artifact groupId="org.nuxeo.lib*" />
+        </includes>
+        <excludes>
+          <artifact scope="test" />
+          <artifact scope="provided" />
+          <artifact type="pom" />
+          <artifact groupId="nuxeo-studio" />
+          <!-- Filter out your artifacts using their groupId for instance -->
```
Comment (jcarsique) [inline marketplace/src/main/assemble/assembly.xml]: please remove comments addressed to customers copying the sample

File: marketplace/src/main/assemble/assembly.xml [hunk 3]
```diff
@@ -1,41 +1,142 @@
...
+    <!-- Optionally attach the NXR for later use by other modules -->
+    <artifact:attach file="${outdir}/nxr-${maven.project.version}.zip" classifier="nxr" type="zip"/>
+
+    <!-- Your MP only needs to include files not already provided by the base distribution -->
+    <nx:zipdiff
+      file1="${outdir}/nxr-${maven.project.version}.zip"
+      file2="${outdir}/nuxeo-nxr-server-${nuxeo.distribution.version}.zip"
+      includesfile="${outdir}/includes"
+      excludesfile="${outdir}/excludesfile"
+      patternsetid="sample.versus.cap"
```
Comment (jcarsique) [inline marketplace/src/main/assemble/assembly.xml]: not `sample.versus.cap`

File: marketplace/src/main/assemble/assembly.xml [hunk 4]
```diff
@@ -1,41 +1,142 @@
...
+      ignoreContent="false"
+      ignoreContentPattern=".*\.jar$"></nx:zipdiff>
+    <unzip src="${outdir}/nxr-${maven.project.version}.zip" dest="${outdir}/marketplace/install">
+      <patternset refid="sample.versus.cap"/>
```
Comment (jcarsique) [inline marketplace/src/main/assemble/assembly.xml]: not `sample.versus.cap`

**Review by jcarsique — CHANGES_REQUESTED** [review]

**Comment by jcarsique** [comment]
:+1: after the few requested changes

**Pull request merged**